AzureArmTool/Commands/NestCommand.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AzureArmTool.Common.FileHandlers;
+using AzureArmTool.Common.Templates;
+using AzureArmTool.Creators;
+using Newtonsoft.Json.Linq;
+
+namespace AzureArmTool.Commands;
+
+internal class NestCommand
+{
+    private readonly List<string> _inputFiles;
+    private readonly string _outputFile;
+
+    public NestCommand(List<string> inputFiles, string outputFile)
+    {
+        _inputFiles = inputFiles;
+        _outputFile = outputFile;
+    }
+
+    public int Run()
+    {
+        try
+        {
+            var inputFilesData = _inputFiles.Select(file =>
+            {
+                using var fileStream = File.OpenRead(file);
+                using var fileReader = new StreamReader(fileStream);
+
+                return (fileName: Path.GetFileName(file).Replace(Path.GetExtension(file), ""), json: fileReader.ReadToEnd());
+            });
+
+            var templates = inputFilesData.Select(data => (data.fileName, template: JObject.Parse(data.json))).ToList();
+
+            if (templates.Count == 0)
+            {
+                throw new InvalidOperationException("No JSON files found.");
+            }
+
+            var templateCreator = new TemplateCreator();
+
+            var linkedTemplate = templateCreator.CreateEmptyTemplate();
+            linkedTemplate.parameters = new Dictionary<string, TemplateParameterProperties>();
+
+            var resources = new List<NestedTemplateResource>();
+            var dependsOn = new List<string>();
+
+            var fileWriter = new FileWriter();
```

Comment:
Denk dat dit een disposable is

Reply:
FileWriter is een internal class met een methode die uiteindelijk File.WriteAllText doet